src/modules/mavlink/mavlink_main.cpp

```diff
@@ -1729,7 +1737,11 @@ Mavlink::configure_streams_to_default(const char *configure_single_stream)
 		configure_stream_local("MAG_CAL_REPORT", 1.0f);
 		configure_stream_local("MANUAL_CONTROL", 5.0f);
 		configure_stream_local("NAV_CONTROLLER_OUTPUT", 10.0f);
+		configure_stream_local("OPEN_DRONE_ID_ARM_STATUS", 1.f);
```

Comment:
Why are we streaming [OPEN_DRONE_ID_ARM_STATUS](https://mavlink.io/en/messages/common.html#OPEN_DRONE_ID_ARM_STATUS)? Its inbound from the remoteID hardware. Should be used to indicate healthy/unhealthy for the remoteID.

Reply:
how much of this behaviour is defined in the spec vs left to the manufacturer to figure out?

Reply:
@tridge may know more about the how the spec has been implemented. The vendor may have just an abstraction layer on top of ODID core library for their specific hardware. Or they may use proprietary software and just satisfy the mavlink ODID definitions and don’t use ardu (or ODID core library)

Reply:
Yeah even in Ardupilot, the `ARM_STATUS` message isn't getting sent out, so I am puzzled as to why Bluemark module needs it 🤔 

https://github.com/ArduPilot/ardupilot/blob/5fa8b887a2df8a85822f84f41eec64bef5066895/libraries/AP_OpenDroneID/AP_OpenDroneID.cpp#L266-L326

Reply:
> Yeah even in Ardupilot, the `ARM_STATUS` message isn't getting sent out, so I am puzzled as to why Bluemark module needs it 🤔
> 
> https://github.com/ArduPilot/ardupilot/blob/5fa8b887a2df8a85822f84f41eec64bef5066895/libraries/AP_OpenDroneID/AP_OpenDroneID.cpp#L266-L326

Correct it is the ArduRemoteID module instead that streams it to PX4.

Reply:
> Why are we streaming [OPEN_DRONE_ID_ARM_STATUS](https://mavlink.io/en/messages/common.html#OPEN_DRONE_ID_ARM_STATUS)? Its inbound from the remoteID hardware. Should be used to indicate healthy/unhealthy for the remoteID.
> 
> I.e. an update to #21652
> 
> I'm probably missing something ...

Correct, but this OPEN_DRONE_ID_ARM_STATUS messages need to be relayed to the GCS software. So the GCS know the Remote ID status. (A mandatory requirement in standard Remote ID.)

Reply:
@BluemarkInnovations I was going to sarcastically say _where did you get that "requirement"_, but you're absolutely right - its in [the MAVLink spec](https://mavlink.io/en/services/opendroneid.html):

> Similarly, the autopilot and GCS components must listen to the [ARM_STATUS](https://mavlink.io/en/services/opendroneid.html#OPEN_DRONE_ID_ARM_STATUS) from the RID transmitter component(s) and not allow the UA to be airborne before the RID transmitter component(s) is ready. During flight, if the arm status indicates a failure, similar action must be taken as for a lack of HEARTBEAT messages.

The vehicle shouldn't start if it doesn't have this so what the GCS does is irrelevant. Though this is needed for monitoring the precise cause of arming failure, unless the flight stack were to emit an event (and none is standardized).

@dirksavage88 What is interesting here is that the arming status is not just a take off interlock. If it is the same as lack of heartbeat then it should be treated the same as data lost safety failsafe: https://docs.px4.io/main/en/config/safety.html#safety-configuration-failsafes

I'll see if I can tidy the spec.

Reply:
Late to the party as I only started looking into Remote ID yesterday, but I had the same initial reaction as @hamishwillee back in June.

In my mind it's more logical if you just set up mavlink forwarding so that the GCS can subscribe directly to the topic from the Remote ID hardware without going through the "useless" steps of repackaging the incoming message and publishing it as uORB only to subscribe to the uORB message to republish a mavlink message with basically the same content as the initial message.

Maybe I am too naive and miss something essential here?

Reply:
Yes, that would make sense, just forwarding it. However, for a module that talks DroneCAN or some other protocol, this might be required again.